src/searchbar.cpp

```diff
@@ -184,11 +184,12 @@ void SearchBarLineEdit::updateCompletion()
     m_token++;
     auto suggestionWorker = new SuggestionListWorker(m_searchbarInput, m_token, 0, this);
     connect(suggestionWorker, &SuggestionListWorker::searchFinished, this,
-    [=] (const QStringList& suggestions, const QVector<QUrl>& urlList, int token) {
+    [=] (const QStringList& suggestions, const QVector<QUrl>& urlList, bool hasFullText, int token) {
```

Comment:
A more straightforward solution (assuming the rest of my design change proposals are implemented) is to introduce a member function `bool SuggestionData::isFullTextSearchSuggestion() const` that works off the contents of the host part of the url (checks if it ends with `.search`).